comp/core/tagger/taggerimpl/server/util.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package server
+
+import (
+	"google.golang.org/protobuf/proto"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+)
+
+// splitBySize splits the given slice into contiguous non-overlapping subslices such that
+// the size of each sub-slice is at most maxChunkSize.
+// The size of each item is calculated using computeSize
+//
+// This function assumes that the size of each single item of the initial slice is not larger than maxChunkSize
+func splitBySize[T any](slice []T, maxChunkSize int, computeSize func(T) int) [][]T {
```

Comment:
declared this generic function in order to make it easy to unit test the split functionality